components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java

```diff
@@ -0,0 +1,313 @@
+                        existingCertificate.getAlias());
+            }
+            String message = "";
+            if (existingCertificate.getAlias().equals(alias)) {
+                message = "Alias";
+            } else {
+                message = "Endpoint";
+            }
+            throw new CertificateManagementException(message);
+        }
+        try {
+            connection = APIMgtDBUtil.getConnection();
+            initialAutoCommit = connection.getAutoCommit();
+            connection.setAutoCommit(false);
+            preparedStatement = connection.prepareStatement(addCertQuery);
+            preparedStatement.setInt(1, tenantId);
+            preparedStatement.setString(2, endpoint);
+            preparedStatement.setString(3, alias);
+            result = preparedStatement.executeUpdate() == 1;
+            connection.commit();
+            connection.setAutoCommit(initialAutoCommit);
+        } catch (SQLException e) {
+            log.error("Error while executing SQL to add certificate.", e);
+            try {
+                if (connection != null) {
+                    connection.rollback();
+                } else {
+                    log.error("Could not perform rollback since the connection is null.");
+                }
+            } catch (SQLException e1) {
+                log.error("Error while rolling back the transaction.", e1);
+            }
+        } finally {
+            APIMgtDBUtil.closeStatement(preparedStatement);
+            APIMgtDBUtil.closeAllConnections(preparedStatement, connection, null);
+        }
+        return result;
+    }
+
+    /**
+     * Method to retrieve certificate metadata from db for specific alias or endpoint.
+     * From alias and endpoint, only one parameter is required. This will be used to query all the certificates
+     * without a limitation for tenant.
+     * <p>
+     * Addresses : If some tenant is trying to add a certificate with the same alias, proper error should be shown in
+     * the UI.
+     *
+     * @param alias    : Alias for the certificate. (Optional)
+     * @param endpoint : The endpoint/ server url which the certificate is mapped to. (Optional)
+     * @return : A CertificateMetadataDTO object if the certificate is retrieved successfully, null otherwise.
+     */
+    private CertificateMetadataDTO getCertificate(String alias, String endpoint) {
+        Connection connection = null;
+        PreparedStatement preparedStatement = null;
+        ResultSet resultSet = null;
+        CertificateMetadataDTO certificateMetadataDTO = null;
+        String getCertQuery = "SELECT * FROM AM_CERTIFICATE_METADATA WHERE (ALIAS=? OR END_POINT=?)";
+
+        try {
+            connection = APIMgtDBUtil.getConnection();
+            connection.commit();
+            preparedStatement = connection.prepareStatement(getCertQuery);
+            preparedStatement.setString(1, alias);
```

Comment:
smae here

Reply:
In apim-6.1.x we could not use java8 syntax.